src/Mvc/Mvc.ApiExplorer/src/DefaultApiDescriptionProvider.cs

```diff
@@ -463,6 +467,20 @@ internal static MediaTypeCollection GetDeclaredContentTypes(IReadOnlyList<IApiRe
                 .ToArray();
         }
 
+        private static string? GetGroupName(ControllerActionDescriptor action)
+        {
+            // The `GroupName` set in the `ApiDescriptionActionData` is either the
+            // group name set via [ApiExplorerSettings(GroupName = "foo")] on the
+            // action or controller. So, this lookup favors the following sequence:
+            // - EndpointGroupName on the action, if it is set
+            // - EndpointGroupName on the controller, if it is set
+            // - ApiExplorerSettings.GroupName on the action, if it is set
+            // - ApiExplorerSettings.GroupName on the controller, if it is set
+            var extensionData = action.GetProperty<ApiDescriptionActionData>();
```

Comment:
You don't need to look this up again since it's already calculated in the caller.